src/target/riscv/riscv.c

```diff
@@ -1614,6 +1614,357 @@ static int riscv_hit_trigger_hit_bit(struct target *target, uint32_t *unique_id)
+		const riscv_insn_t instruction, target_addr_t *mem_addr_wp) {
+	/* find out which memory address is accessed by the instruction at dpc */
+	/* opcode is first 7 bits of the instruction */
+	uint32_t opcode = instruction & INSN_FIELD_OPCODE;
+	if ((instruction & 0x03) < 0x03) { // opcode size RVC
+		/* RVC MASK_C = 0xe003 for load/store instructions */
+		opcode = instruction & MASK_C_LD;
+	}
+	uint32_t rs1;
+	int16_t imm;
+	target_addr_t mem_addr = 0;
+	bool misa_f = riscv_supports_extension(target, 'F');
+	bool misa_d = riscv_supports_extension(target, 'D');
+
+	switch (opcode) {
+	case MATCH_LB:
+		/* fallthrough */
+	case MATCH_FLH & ~INSN_FIELD_FUNCT3:
+		/* fallthrough */
+	case MATCH_SB:
+		/* fallthrough */
+	case MATCH_FSH & ~INSN_FIELD_FUNCT3:
+		rs1 = get_field32(instruction, INSN_FIELD_RS1);
+		if (riscv_get_register(target, &mem_addr, rs1) != ERROR_OK)
+			return ERROR_FAIL;
+		if (opcode == MATCH_SB || opcode == (MATCH_FSH & ~INSN_FIELD_FUNCT3)) {
+			LOG_TARGET_DEBUG(target, "%x is store instruction", instruction);
+			imm = get_field32(instruction, INSN_FIELD_IMM12LO) |
+				  (get_field32(instruction, INSN_FIELD_IMM12HI) << 5);
+		} else if (opcode == MATCH_LB ||
+				   opcode == (MATCH_FLH & ~INSN_FIELD_FUNCT3)) {
+			LOG_TARGET_DEBUG(target, "%x is load instruction", instruction);
+			imm = get_field32(instruction, INSN_FIELD_IMM12);
+		} else {
+			return ERROR_FAIL;
+		}
+		/* sign extend 12-bit imm to 16-bits */
+		if (imm & (1 << 11))
+			imm |= 0xf000;
+		mem_addr += imm;
+		LOG_TARGET_DEBUG(target, "Memory address=0x%" PRIx64, mem_addr);
+		break;
+
+	case MATCH_C_LWSP:
+		if (get_field32(instruction, INSN_FIELD_RD) == 0) // the code points with rd=x0 are reserved
+			return ERROR_FAIL;
+		if (riscv_get_register(target, &mem_addr, GDB_REGNO_SP) != ERROR_OK)
+			return ERROR_FAIL;
+		imm = get_imm_clwsp(instruction);
+		mem_addr += imm;
+		LOG_TARGET_DEBUG(target, "C.LWSP Memory address=0x%" PRIx64, mem_addr);
+		break;
+
+	case MATCH_C_LDSP: // if xlen >= 64 or MATCH_C_FLWSP:
+		if (riscv_xlen(target) > 32) { // MATCH_C_LDSP
+			if (get_field32(instruction, INSN_FIELD_RD) == 0) // the code points with rd=x0 are reserved
+				return ERROR_FAIL;
+			if (riscv_get_register(target, &mem_addr, GDB_REGNO_SP) != ERROR_OK)
+				return ERROR_FAIL;
+			imm = get_imm_cldsp(instruction);
+			mem_addr += imm;
+			LOG_TARGET_DEBUG(target, "C.LDSP memory address=0x%" PRIx64,
+							 mem_addr);
+		} else { // MATCH_C_FLWSP
+			if (!misa_f)
+				return ERROR_FAIL;
+			if (riscv_get_register(target, &mem_addr, GDB_REGNO_SP) != ERROR_OK)
+				return ERROR_FAIL;
+			imm = get_imm_clwsp(instruction);
+			mem_addr += imm;
+			LOG_TARGET_DEBUG(target, "C.FLWSP memory address=0x%" PRIx64,
+							 mem_addr);
+		}
+		break;
+
+	case MATCH_C_FLDSP: // or MATCH_C_LQSP if RV128C
+		if (riscv_xlen(target) == 128) { // MATCH_C_LQSP
+			if (get_field32(instruction, INSN_FIELD_RD) == 0) // the code points with rd=x0 are reserved
```

Comment:
Here and around: please, consider moving comments, describing why the instruction is not load/store to a LOG_TARGET_DEBUG. This will make logs much clearer.
```suggestion
			if (get_field32(instruction, INSN_FIELD_RD) == 0)
			     LOG_TARGET_DEBUG(target, "The code points with rd=x0 are reserved in case of MATCH_C_LQSP.");
```

Reply:
Added